legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSQLQueryAction.java

```diff
@@ -61,6 +65,14 @@ public class RestSQLQueryAction extends BaseRestHandler {
    */
   private final Settings pluginSettings;
 
+  /**
+   * Captured error message to aggregate diagnostics
+   * for both legacy and new SQL engines.
+   * This member variable and it's usage can be deleted once the
+   * legacy SQL engine is deprecated.
+   */
+  private String errorStr;
```

Comment:
You can use `@Getter` and `@Setter` annotations instead of creating `setErrorStr` and `getErrorStr`.

Reply:
Resolved under: [08981598f03df7da85f23fd547b70c1a08318c8c](https://github.com/opensearch-project/sql/pull/691/commits/08981598f03df7da85f23fd547b70c1a08318c8c)

Reply:
Just want to confirm is this class (or `RestSqlAction` who create this) a singleton or created per request?

Reply:
do we really need errorStr? or Exception e? Could we use RestSqlAction produce the error message from e.

Reply:
@dai-chen This class is a singleton.
@penghuo We need to be able to capture error messages form potentially 2 exceptions. That is why we need the `errorStr` member variable.

Reply:
RestSqlAction is created per request.